packages/transformers/webmanifest/src/WebManifestTransformer.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+// https://developer.mozilla.org/en-US/docs/Web/Manifest
+import type {MutableAsset} from '@parcel/types';
+import type {SchemaEntity} from '@parcel/utils';
+
+import {Transformer} from '@parcel/plugin';
+import invariant from 'assert';
+import {parse} from 'json-source-map';
+import {getJSONSourceLocation} from '@parcel/diagnostic';
+import {validateSchema} from '@parcel/utils';
+
+type Manifest = {
+  serviceworker?: string,
+  icons: Array<{src: string, ...}>,
+  screenshots: Array<{src: string, ...}>,
+  ...
+};
+
+const ICONS_SCHEMA = {
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      src: {
+        type: 'string',
+        __validate: s => {
+          if (s.length === 0) {
+            return 'Must not be empty';
+          }
+        },
+      },
+    },
+    required: ['src'],
+  },
+};
+const MANIFEST_SCHEMA: SchemaEntity = {
+  type: 'object',
+  properties: {
+    icons: ICONS_SCHEMA,
+    screenshots: ICONS_SCHEMA,
+  },
+};
+
+function collectDependencies(
+  asset: MutableAsset,
+  source: string,
+  data: Manifest,
+  pointers: any,
+) {
+  validateSchema.diagnostic(
+    MANIFEST_SCHEMA,
+    {source, map: {data, pointers}, filePath: asset.filePath},
+    '@parcel/transformer-webmanifest',
+    'Invalid webmanifest',
+  );
+
+  for (let key of ['icons', 'screenshots']) {
+    let list = data[key];
+    if (list) {
+      invariant(Array.isArray(list));
+      for (let i = 0; i < list.length; i++) {
+        let icon = list[i];
```

Comment:
I reckon this would be better named `resource` over `icon`.

Trying this locally I went with

```js
function collectDependencies(asset, source, data, pointers) {
  return [ "icons", "screenshots" ].forEach((resourceType) => {
    ;(data[resourceType] || []).forEach((resource, i) => {
      resource.src = asset.addURLDependency(resource.src, {
        loc: {
          filePath: asset.filePath,
          ...getJSONSourceLocation(pointers[`/${resourceType}/${i}/src`], "value"),
        },
      })
    })
  })
}
```

Reply:
This could be related to the `addURLDependency` vs `addDependency`? Trying to use `url:./icons/180.png` didn't compile the files or interpolate the URI. Using `./icons/180.png` compiled the files but the URI is the hash seen above.